### ISSUE TITLE
Rename Repository using Definitions

### ISSUE THREAD

**Issue by tnk-dev** [dagster-io/dagster#11424]
### What's the use case?

Defining the repository name in the workspace.yaml,  takes away the flexibility to use python for dynamic names. For e.g. we append the current version of our build
```python
@repository(
    name=f'data_platform_repo_v{get_current_version()}_{STACK}',
    ...
```
Same goes for `description`.

Is it possible for you guys to add this to your roadmap to include name and description arguments to Definition?

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by schrockn** [comment]
We generally don't recommend dynamically naming repositories in this fashion. Names in dagster are meant to be stable over time as they are the targets of scripts and consistent names are critical for many of our operational tools.

If you need this capability you can continue to use `@repository` which will continue to be supported for the foreseeable future.

If you want to use a more Definitions-esque API, the next release will include `create_repository_using_definitions_args` in `dagster._core.definitions.definitions_class` that can create named repositories with same API semantics as `Definitions`.

**Comment by tnk-dev** [comment]
I agree that dynamically named repos aren't that smart, so we removed that dynamic naming. I see `create_repository_using_definitions_args` as of 1.1.8 allows naming the repo now.

What is still missing for me is a **repository description** shown in Dagit, where all kinds of metadata about the repo, etc could be displayed.
